src/service/service.ts

```diff
@@ -740,7 +740,12 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
   private handleFindNode(nodeAddr: INodeAddress, message: IFindNodeMessage): void {
     const { id, distances } = message;
     let nodes: ENR[] = [];
-    distances.forEach((distance) => {
+    let distinctDistances = distances.filter((n, i) => distances.indexOf(n) === i);
```

Comment:
maybe better to just convert to a set to filter out duplicates
```ts
for (const distance of new Set(distances)) {
  // filter out invalid distances
  ...
```

Reply:
Oh that looks much better. Thank you. Will fix.